### PR TITLE
Fix position embeddings singular/plural

### PR DESCRIPTION
**Pull request by molbap**
# What does this PR do?

It seems a few Blip-derived models have both `self.position_embedding` and `self.position_embeddings`, plural, defined in the same module and causing some errors. This aims at fixing those. Noticed it in #31368 .

Causes many things to break because self.position_embedding isn't defined since it's self.position_embeddings (with an s) instead.

### CODE REVIEW

**Comment by molbap** [comment]
cc @amyeroberts for review!

**Comment by molbap** [comment]
@amyeroberts sorry for the ping, just keeping this one higher in the notifications - hub tests were just flaky :)

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing!

File: src/transformers/models/blip_2/modeling_blip_2.py [hunk 1]
```diff
@@ -200,7 +200,6 @@ def __init__(self, config: Blip2VisionConfig):
 
         self.position_embedding = nn.Parameter(torch.randn(1, self.num_positions, self.embed_dim))
 
-    # Copied from transformers.models.vit.modeling_vit.ViTEmbeddings.interpolate_pos_encoding
```
[the remark this answers is not in the document]
Comment (molbap) [reply src/transformers/models/blip_2/modeling_blip_2.py:203]: Ah, good point! adding

[the remark this answers is not in the document]
Comment (molbap) [reply src/transformers/models/blip_2/modeling_blip_2.py:203]: uh, actually I missed it - the whole `Blip2VisionEmbeddings` class is already copied from `BlipVisionEmbeddings`. Seems that writing a specific method of that class to be also a copy of the original class method breaks `fix-copies`.

[the remark this answers is not in the document]
Comment (amyeroberts) [reply src/transformers/models/blip_2/modeling_blip_2.py:203]: Ah, I see, makes sense why this statement is removed everywhere then :)

[the remark this answers is not in the document]
Comment (molbap) [reply src/transformers/models/blip_2/modeling_blip_2.py:203]: Yes! Merging, then :)

**Comment by molbap** [comment]
Found out about an issue in `test_expansion_in_processing` for `instructblip` @zucchini-nlp and fixed it here, not a big deal I think but FYI (list + tensor concat messing expansion!)

**Pull request merged**